### PR TITLE
enable inline actions for connection dialog's browse tab

### PR DESCRIPTION
**Pull request by alanrenmsft**
This is part of the browse connection feature GA work.

There is a feature gap for browse Azure resources between the connections view and connection dialog, currently the actions are only available as context menu in the connection dialog, in this PR, I am adding support for inline action bar, just like in the connections view, the actions are only visible when the tree item is focused/selected/hover.

In Connections View:
image

In Connection Dialog:
Before:
image

After:
image
image

### CODE REVIEW

**Comment by abist** [comment]
Will these show up for all kinds of connections such as CMS?

**Review by abist — APPROVED** [review]

**Comment by alanrenmsft** [comment]
> Will these show up for all kinds of connections such as CMS?

@abist this is for saved connections, so CMS won't show up there, right?

**Comment by abist** [comment]
@alanrenmsft Yeah, it won't. I thought this was for all connections.

File: src/sql/workbench/services/connection/browser/connectionBrowseTab.ts [hunk 1]
```diff
@@ -150,9 +150,17 @@ export class ConnectionBrowserView extends Disposable implements IPanelView {
 	renderTree(container: HTMLElement): void {
 		this.treeContainer = container.appendChild(DOM.$('div'));
 		this.treeLabels = this._register(this.instantiationService.createInstance(ResourceLabels, this));
+		this.treeMenus = this.instantiationService.createInstance(TreeMenus);
+		this.treeMenus.contextKeyService = this.contextKeyService;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionBrowseTab.ts:154]: Why not just have the service passed in via the instantiation service? Or at the very least as an actual parameter to the constructor.

File: src/sql/workbench/services/connection/browser/connectionBrowseTab.ts [hunk 2]
```diff
@@ -538,21 +555,67 @@ interface ITreeExplorerTemplateData {
 	container: HTMLElement;
 	resourceLabel: IResourceLabel;
 	icon: HTMLElement;
-	// actionBar: ActionBar;
+	actionBar: ActionBar;
+}
+
+class TreeMenus {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionBrowseTab.ts:561]: This name seems a little too generic for something that's pretty specific to the browse tree and it's also not really a "menu" itself. Would suggest something else like `ConnectionBrowseTreeMenuProvider` or similar

**Review by Charles-Gagnon — APPROVED** [review]

File: src/sql/workbench/services/connection/browser/connectionBrowseTab.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/workbench/services/connection/browser/connectionBrowseTab.ts:154]: 😄 I don't know what I was thinking.

File: src/sql/workbench/services/connection/browser/connectionBrowseTab.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/workbench/services/connection/browser/connectionBrowseTab.ts:561]: this is a local class and won't be exported, but sure, I can update the name

**Pull request merged**